КР 2.3 №1/List1.c

```diff
@@ -0,0 +1,100 @@
+﻿#include <stdio.h>
+#include <stdlib.h>
+#include "List.h"
+
+
+struct Node {
+	int value;
+	Node* next;
+};
+
+struct List {
+	Node* head = NULL;
+};
+
+List* createList() {
+	return new List;
+}
+
+bool isEmpty(List* list) {
+	return list->head == NULL;
+}
+
+void addValue(List* list, int value) {
+	Node* current = list->head;
+	Node* previous = NULL;
+	while (current != NULL && current->value < value) {
+		previous = current;
+		current = current->next;
+	}
+	Node* newNode = new Node{ value, current };
+	if (previous == NULL) {
+		list->head = newNode;
+	}
+	else {
+		previous->next = newNode;
+	}
+}
+
+bool deleteValue(List* list, int value) {
+	Node* current = list->head;
+	Node* previous = NULL;
+	while (current != NULL && current->value != value) {
+		previous = current;
+		current = current->next;
+	}
+	if (current == NULL) {
+		return false;
+	}
+	Node* temp = current->next;
+	free (current);
```

Comment:
```suggestion
	free(current);
```
Кстати, вообще new/delete и malloc/free могут выделять память по-разному, так что по стандарту использовать их совместно нельзя :)